gradoop-temporal/src/main/java/org/gradoop/temporal/io/impl/csv/indexed/TemporalIndexedCSVDataSink.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright © 2014 - 2020 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.temporal.io.impl.csv.indexed;
+
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.apache.flink.core.fs.FileSystem.WriteMode;
+import org.apache.flink.core.fs.Path;
+import org.gradoop.flink.io.impl.csv.CSVConstants;
+import org.gradoop.flink.io.impl.csv.indexed.IndexedCSVDataSink;
+import org.gradoop.flink.io.impl.csv.indexed.functions.IndexedCSVFileFormat;
+import org.gradoop.flink.io.impl.csv.metadata.CSVMetaDataSink;
+import org.gradoop.flink.io.impl.csv.metadata.CSVMetaDataSource;
+import org.gradoop.flink.util.GradoopFlinkConfig;
+import org.gradoop.temporal.io.api.TemporalDataSink;
+import org.gradoop.temporal.io.impl.csv.functions.TemporalEdgeToTemporalCSVEdge;
+import org.gradoop.temporal.io.impl.csv.functions.TemporalGraphHeadToTemporalCSVGraphHead;
+import org.gradoop.temporal.io.impl.csv.functions.TemporalVertexToTemporalCSVVertex;
+import org.gradoop.temporal.io.impl.csv.tuples.TemporalCSVEdge;
+import org.gradoop.temporal.io.impl.csv.tuples.TemporalCSVGraphHead;
+import org.gradoop.temporal.io.impl.csv.tuples.TemporalCSVVertex;
+import org.gradoop.temporal.model.impl.TemporalGraph;
+import org.gradoop.temporal.model.impl.TemporalGraphCollection;
+
+import java.io.IOException;
+
+/**
+ * A temporal graph data sink for CSV files indexed by label.
+ */
+public class TemporalIndexedCSVDataSink extends IndexedCSVDataSink implements TemporalDataSink {
```

Comment:
You should not extend `IndexedCSVDataSink`, since you aren't actually using or overriding(!) any of the write methods. Better to extend `CSVBase` directly. If you wanted to reuse code, you would have to create a generified `BaseIndexedCSVDataSink` and extend it for the normal and the temporal indexed csv data sink.